pkg/render/dex_utils.go

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package render
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	oprv1 "github.com/tigera/operator/api/v1"
+)
+
+type ConnectorType string
+
+const (
+	ConnectorTypeOIDC      = "oidc"
+	ConnectorTypeOpenshift = "openshift"
+	ConnectorTypeGoogle    = "google"
+)
+
+type DexConfig interface {
+	// BaseURL returns the address where the Manager UI can be found. Ex: https://example.org
+	BaseURL() string
+	// UsernamePrefix returns the string to prepend to every username for RBAC.
+	UsernamePrefix() string
+	// GroupsPrefix returns the string to prepend to every group for RBAC.
+	GroupsPrefix() string
+	// UsernameClaim returns the part of the JWT that represents a unique username.
+	UsernameClaim() string
+	// GroupsClaim returns the part of the JWT that represents the list of user groups.
+	GroupsClaim() string
+	// ClientSecret returns the secret for Dex' auth endpoint
+	ClientSecret() []byte
+	TLSSecret() *corev1.Secret
+	IdpSecret() *corev1.Secret
+	DexSecret() *corev1.Secret
+	GoogleServiceAccountSecret() []byte
+	OpenshiftRootCA() []byte
+
+	// AppendDexEnv adds env that is used to configure pods with dex options.
+	AppendDexEnv(env []corev1.EnvVar, prefix string) []corev1.EnvVar
```

Comment:
Could you get rid of the "Append" functions and just return the env list please? This appending logic doesn't really help anything.